app/src/main/java/com/duckduckgo/app/onboarding/ui/OnboardingActivity.kt

```diff
@@ -38,7 +42,9 @@ class OnboardingActivity : DuckDuckGoActivity() {
 
     fun onContinueClicked() {
         val next = viewPager.currentItem + 1
-        if (next < viewPager.adapter!!.count) {
+        if (next < viewPager.adapter!!.count &&
+            !variantManager.getVariant().hasFeature(VariantManager.VariantFeature.SetDefaultBrowserDialog)
```

Comment:
Correct me if I'm wrong, but I think this condition here assumes that between the First Onboarding screen and the browser Activity there's only 1 screen, and that screen can be skipped when feature `SetDefaultBrowserDialog` is true. 

I think we should avoid having this condition here. What about just not adding pages to the adapter based on that variant feature, and then we can just keep the logic as it was?

Reply:
Agree. I will move the logic into the `DefaultBrowserDetector`